sqlmesh/core/engine_adapter/spark.py

```diff
@@ -260,8 +264,14 @@ def _df_to_source_queries(
 
         def query_factory() -> Query:
             temp_table = self._get_temp_table(target_table or "spark", table_only=True)
-            df.createOrReplaceGlobalTempView(temp_table.sql(dialect=self.dialect))  # type: ignore
-            temp_table.set("db", "global_temp")
+            if self.use_serverless:
```

Comment:
This looks like databricks specific. Should we just override `query_factory` it in the corresponding adapter?

Reply:
EG.:
```
def query_factory() -> Query:
            temp_table = self._get_temp_table(target_table or "spark", table_only=True)
            if self.use_serverless:
                ...
            else:
                super()....
```

Reply:
Hmm I think I will just completely override it since the `_use_spark_session` is also databricks specific. 

Change: https://github.com/TobikoData/sqlmesh/pull/3001/commits/91cb2a16e058fa297f1e7985d81fadd1f104a627